noir-projects/aztec-nr/aztec/src/utils/bytes.nr

```diff
@@ -70,19 +78,26 @@ pub fn le_bytes_31_to_fields<let N: u32>(bytes: [u8; N]) -> [Field; (N + 30) / 3
 }
 
 /// Converts an input array of fields into a single array of bytes.
-/// Each input field must contain only 31 bytes (this is constrained to be so).
-/// Each field is converted into 31 big-endian bytes, and the resulting 31-byte
-/// chunks are concatenated back together in the order of the original fields.
+/// Each input field must contain at most 31 bytes (this is constrained to be so).
+/// Note that the last field might contain less than 31 bytes. This is communicated through the M parameter.
+///
+/// We have the M parameter be flexible instead of defined with arithmetic over generic because it's common that
+/// this function is used to convert fields to bytes where the input was previously converted from bytes to fields,
+/// while the original input did not necessarily have a length that is a multiple of 31. Having the M parameter be
+/// flexible allows us to get the byte array that is the same length as the original input.
+///
+/// Each field (other than the last which might be shorter) is converted into 31 big-endian bytes, and the
+/// resulting 31-byte chunks are concatenated back together in the order of the original fields. The bytes within
+/// each field are ordered from most significant to least significant.
```

Comment:
This comment was updated as I felt it makes sense to expand on why the M param is flexible and instead of the Mike's note:
"It appears that the original author of this function wants to give the caller some control"

I just put the relevant info next to the assert.